e2e/jenkins/crw-ci/k8s/pr-check/Jenkinsfile

```diff
@@ -0,0 +1,229 @@
+"""
+                        }
+                    }
+                }
+
+                stage("Start Kubernetes") {
+                    steps {
+                        script {
+                            sh """
+  /usr/local/bin/minikube start \\
+    --vm-driver=none \\
+    --cpus 4 \\
+    --memory 12288 \\
+    --logtostderr
+
+  sudo chmod g+r -R /home/codenvy/.minikube
+  sudo chmod g+r -R /home/codenvy/.kube
+  sudo chmod g+r /var/lib/kubeadm.yaml
+"""
+                        }
+                    }
+                }
+
+                stage("Build Che") {
+                    steps {
+                        script {
+                            sh "mvn clean install -f ${WORKSPACE}/che/pom.xml"
+                        }
+                    }
+                }
+            }
+        }
+
+        stage("Push Che artifacts") {
+            steps {
+                withCredentials([string(credentialsId: 'ed71c034-60bc-4fb1-bfdf-9570209076b5', variable: 'maxura_docker_password')]) {
+                    script {
+                        sh """
+  ${WORKSPACE}/che/dockerfiles/che/build.sh --organization:maxura --tag:${CHE_IMAGE_TAG} --dockerfile:Dockerfile
+  docker tag ${CHE_IMAGE_REPO}:${CHE_IMAGE_TAG} docker.io/${CHE_IMAGE_REPO}:${CHE_IMAGE_TAG}
+  docker login -u maxura -p ${maxura_docker_password}
+  docker push docker.io/${CHE_IMAGE_REPO}:${CHE_IMAGE_TAG}
+"""
+                    }
+
+                }
+            }
+        }
+
+        stage("Start Che") {
+            steps {
+                script {
+                    sh """
+  /usr/local/bin/chectl server:start \\
+   --k8spodreadytimeout=180000 \\
+   -t=${WORKSPACE}/deploy/ \\
+   --listr-renderer=verbose \\
+   --cheimage=${CHE_IMAGE_REPO}:${CHE_IMAGE_TAG}
+"""
+
+                    // wait che-server to be available
+                    sh """
+     CHE_URL=\$(kubectl get ingress che-ingress -n=che -o=jsonpath={'.spec.rules[0].host'})
+     
+     COUNTER=0;
+     SUCCESS_RATE_COUNTER=0;
+     while true; do
+      if [ \$COUNTER -gt 180 ]; then
+      echo "Unable to get stable route. Exiting"
+        exit 1
+      fi
+      
+      ((COUNTER+=1))
+      
+      
+      STATUS_CODE=\$(curl -sL -w "%{http_code}" -I \${CHE_URL} -o /dev/null; true) || true # added true to not fail the script completely.
+      
+      echo "Try \${COUNTER}. Status code: \${STATUS_CODE}"
+      if [ "\$STATUS_CODE" == "200" ]; then 
+        ((SUCCESS_RATE_COUNTER+=1))
+      fi
+      sleep 1;
+    
+      if [ \$SUCCESS_RATE_COUNTER == \$SUCCESS_THRESHOLD ]; then 
+        echo "Route is stable enough. Continuing running tests"
+        break
+      fi
+     done
+"""
+                }
+            }
+        }
+
+        stage("Create ws from devfile") {
+            steps {
+                script {
+                    sh """
+  /usr/local/bin/chectl workspace:start --devfile=$DEVFILE_URL
+  kubectl get configmaps --namespace=che che -o yaml
```

Comment:
Some debugging leftovers?

Reply:
It could be moved to post action step, indeed.